app/models/accounting/quickbooks/full_fetcher.rb

```diff
@@ -26,23 +23,40 @@ def fetch_all
         qb_connection.update_attribute(:last_updated_at, started_fetch_at)
 
         restore_accounts!(accounts)
+       rescue StandardError => error
+         delete_qb_data
```

Comment:
I'm uncertain about doing this much in a rescue block that overlaps with what we're rescuing from. This sort of assumes the error is going to be in the 'fetching' stage.

Reply:
I can refactor so that rescuing from fetching is separate, and deletes data as well as cnxn.